### PR TITLE
change the esd, arts and nas audio backends be disabled by default.

### PR DESCRIPTION
**Pull request by sezero**
Don't know anyone still use these ones. Their options are still there, but disabled by default.

CC: @madebr for cmake side.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -455,15 +455,15 @@ set_option(SDL_ALSA                "Support the ALSA audio API" ${UNIX_SYS})
 dep_option(SDL_ALSA_SHARED         "Dynamically load ALSA audio support" ON "SDL_ALSA" OFF)
 set_option(SDL_JACK                "Support the JACK audio API" ${UNIX_SYS})
 dep_option(SDL_JACK_SHARED         "Dynamically load JACK audio support" ON "SDL_JACK" OFF)
-set_option(SDL_ESD                 "Support the Enlightened Sound Daemon" ${UNIX_SYS})
+set_option(SDL_ESD                 "Support the Enlightened Sound Daemon" OFF ${UNIX_SYS} OFF)
```
Comment (madebr) [inline CMakeLists.txt:458]: `set_option` is an unconditional option.
`dep_option` is a conditional option.
```suggestion
det_option(SDL_ESD                 "Support the Enlightened Sound Daemon" OFF ${UNIX_SYS} OFF)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -455,15 +455,15 @@ set_option(SDL_ALSA                "Support the ALSA audio API" ${UNIX_SYS})
...
 dep_option(SDL_ESD_SHARED          "Dynamically load ESD audio support" ON "SDL_ESD" OFF)
 set_option(SDL_PIPEWIRE            "Use Pipewire audio" ${UNIX_SYS})
 dep_option(SDL_PIPEWIRE_SHARED     "Dynamically load Pipewire support" ON "SDL_PIPEWIRE" OFF)
 set_option(SDL_PULSEAUDIO          "Use PulseAudio" ${UNIX_SYS})
 dep_option(SDL_PULSEAUDIO_SHARED   "Dynamically load PulseAudio support" ON "SDL_PULSEAUDIO" OFF)
-set_option(SDL_ARTS                "Support the Analog Real Time Synthesizer" ${UNIX_SYS})
+set_option(SDL_ARTS                "Support the Analog Real Time Synthesizer" OFF ${UNIX_SYS} OFF)
```
Comment (madebr) [inline CMakeLists.txt:464]:
```suggestion
det_option(SDL_ARTS                "Support the Analog Real Time Synthesizer" OFF ${UNIX_SYS} OFF)
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -455,15 +455,15 @@ set_option(SDL_ALSA                "Support the ALSA audio API" ${UNIX_SYS})
...
 dep_option(SDL_ARTS_SHARED         "Dynamically load aRts audio support" ON "SDL_ARTS" OFF)
-set_option(SDL_NAS                 "Support the NAS audio API" ${UNIX_SYS})
+set_option(SDL_NAS                 "Support the NAS audio API" OFF ${UNIX_SYS} OFF)
```
Comment (madebr) [inline CMakeLists.txt:466]:
```suggestion
det_option(SDL_NAS                 "Support the NAS audio API" OFF ${UNIX_SYS} OFF)
```

File: CMakeLists.txt (hunk 1 above)
Comment (sezero) [reply CMakeLists.txt]: Done - I hope, please confirm.

Comment (madebr) [reply CMakeLists.txt]: Looks okay now!

**Comment by slouken** [comment]
Why would you change this?

**Comment by sezero** [comment]
ESD, for e.g., is not installed by default on distros nowadays, i.e. they aren't built by default already. It's annoying to see the `esd-config` not found like warning messages from configuration.

**Comment by slouken** [comment]
I'd rather not break someone who's potentially using this somewhere on their system. We can remove older backends in SDL3 and gauge the impact then.

**Comment by icculus** [comment]
These are _definitely_ going away in SDL3, but yes, let's leave them for SDL2, the same way we didn't remove Mac OS 9 support from SDL 1.2.

**Pull request closed without merging**